source/slang/slang-ir-lower-generics.cpp

```diff
@@ -45,6 +46,20 @@ namespace Slang
             workListSet.Add(inst);
         }
 
+        bool isPolymorphicType(IRInst* typeInst)
+        {
+            if (as<IRParam>(typeInst) && as<IRTypeType>(typeInst->getFullType()))
+                return true;
+            switch (typeInst->op)
+            {
+            case kIROp_AssociatedType:
```

Comment:
It seems like `IRAssociatedType` is the case causing problems for linking, but it doesn't seem like any code outside the module that defined an interface needs to refer to this representation of an associated type; it exists only to allow us to put signatures on interface requirements.